libs/ledger-live-common/src/families/elrond/js-estimateMaxSpendable.ts

```diff
@@ -19,22 +21,61 @@ const estimateMaxSpendable = async ({
   parentAccount: Account | null | undefined;
   transaction: Transaction | null | undefined;
 }): Promise<BigNumber> => {
-  const a = getMainAccount(account, parentAccount) as ElrondAccount;
-  const t = {
+  const mainAccount = getMainAccount(account, parentAccount);
+  const tx = {
     ...createTransaction(),
+    subAccountId: account.type === "Account" ? null : account.id,
     ...transaction,
-    amount: a.spendableBalance,
   };
-  const fees = await getEstimatedFees({
-    a,
-    t,
-  });
 
-  if (fees.gt(a.spendableBalance)) {
+  const tokenAccount =
+    tx.subAccountId &&
+    mainAccount.subAccounts &&
+    mainAccount.subAccounts.find((ta) => ta.id === tx.subAccountId);
+
+  if (tokenAccount) {
+    return tokenAccount.balance;
+  }
+
+  switch (tx?.mode) {
+    case "reDelegateRewards":
+      tx.gasLimit = GAS.DELEGATE;
+
+      tx.data = ElrondEncodeTransaction.reDelegateRewards();
+      break;
+    case "withdraw":
+      tx.gasLimit = GAS.DELEGATE;
+
+      tx.data = ElrondEncodeTransaction.withdraw();
+      break;
+    case "unDelegate":
+      tx.gasLimit = GAS.DELEGATE;
+
+      tx.data = ElrondEncodeTransaction.unDelegate(tx);
+      break;
+    case "delegate":
+      tx.gasLimit = GAS.DELEGATE;
+
+      tx.data = ElrondEncodeTransaction.delegate();
+      break;
+
+    case "claimRewards":
+      tx.gasLimit = GAS.CLAIM;
+
+      tx.data = ElrondEncodeTransaction.claimRewards();
+      break;
+
+    default:
+      break;
+  }
```

Comment:
Yes, it totally make sense since calculates fees for any type of transaction, not just move balance.
I adapted the `getEstimatedFees` function to take into consideration the transaction `gasLimit` and `data` in order to make sure that all fees are correct calculated.